Class/Practise 3.1/src/test.cpp

```diff
@@ -0,0 +1,81 @@
+#include <iostream>
+#include "Helpers.hpp"
+#include "UnitTests.hpp"
+
+int main() {
+    UnitTests testSuite;
+
+    testSuite.addTest("ZeroInputValueReturnsZero",
+    [](){
+        // Build:
+        double value = 0.0;
+    
+        // Operate:
+        auto result = dummyFunc(value);
+
+        // Check:
+        ASSERT_EQ(result, 0.0);
+    });
+
+    testSuite.addTest("PositiveValueInputReturnsPositiveValue",
+    [](){
+        // Build:
+        double value = 5.0;
+    
+        // Operate:
+        auto result = dummyFunc(value);
+
+        // Check:
+        ASSERT_EQ(result, 5.0);
+    });
+
+    testSuite.addTest("NegativeValueInputReturnsZero",
+    [](){
+        // Build:
+        double value = -3.0;
+    
+        // Operate:
+        auto result = dummyFunc(value);
+
+        // Check:
+        ASSERT_EQ(result, 0.0);
+    });
+
+     // Tests for repeatString
+    testSuite.addTest("RepeatStringPrints_home_Twice", []() {
+        std::vector<int> vec = {0, 2, 1};
+        std::string str = "home";
+        auto result = repeatString(vec, str);
```

Comment:
It is better to add empty lines between logical blocks Build-Operate-Check or Arrange-Act-Assert. Even if you will omit the comment with block name empty lines will make the trick. Otherwise it is harder to read the code.

Also it would be great to have some unit test with empty vector and another one with empty string as an input to cover corner cases

Reply:
Refactored code in https://github.com/YaroslavKSE/Professinal-Programming/pull/7/commits/eb553f5aca2ab802bbcab07b0f76848c99af5285